crates/ruff_cli/src/commands/check.rs

```diff
@@ -119,7 +120,16 @@ pub(crate) fn check(
                         }
                     });
 
-                    lint_path(path, package, &settings.linter, cache, noqa, fix_mode).map_err(|e| {
+                    lint_path(
+                        path,
+                        package,
+                        &settings.linter,
+                        cache,
+                        noqa,
+                        fix_mode,
+                        settings.unsafe_fixes,
```

Comment:
I think accessing from `settings` here could lead to inconsistencies, since the Printer and Emitter and such will use the "global" value, but accessing from `settings` here will read the `unsafe-fixes` from a nested configuration file -- if I understand correctly.

For example... if you had `unsafe-fixes = true` in a `pyproject.toml` at the top of the project, and then `unsafe-fixes = false` in a `pyproject.toml` in a subdirectory, wouldn't the CLI _think_ you ran with `unsafe-fixes`, where internally you would _not_ run with `unsafe-fixes` when analyzing files in the subdirectory?

I suspect this should be passed down from the top-level like `fix_mode` for consistency throughout the execution.

Reply:
Hm okay let me investigate that

Reply:
Yeah you're right! Good catch — I misunderstood that call hierarchy.